corehq/apps/aggregate_ucrs/column_specs.py

```diff
@@ -308,27 +310,23 @@ def to_sqlalchemy_query_column(self, sqlalchemy_table, aggregation_params):
         return self.sqlalchemy_fn(sqlalchemy_table.c[self.properties.referenced_column])
 
 
-class SumColumnAdapter(SimpleAggregationAdapater):
-    """
-    A SimpleAggregationAdapater that sums the values of a given column in the secondary table.
-    """
-    @property
-    def sqlalchemy_fn(self):
-        return sqlalchemy.func.sum
-
-    def get_datatype(self):
-        return DATA_TYPE_INTEGER
+class SqlaggColumnAdapter(SimpleAggregationAdapater):
 
+    def _get_sqlagg_column(self):
+        return SQLAGG_COLUMN_MAP[self._db_column.aggregation_type](self._db_column.column_id)
 
-class MinColumnAdapter(SimpleAggregationAdapater):
-    """
-    A SimpleAggregationAdapater that returns the min value of a given column in the secondary table.
-    """
     @property
     def sqlalchemy_fn(self):
-        return sqlalchemy.func.min
+        return self._get_sqlagg_column().aggregate_fn
 
     def get_datatype(self):
+        # special case some columns to allow to e.g. count unique values from a string column
+        if self._db_column.aggregation_type in (
+                const.AGGGREGATION_TYPE_COUNT,
+                const.AGGGREGATION_TYPE_COUNT_UNIQUE
+        ):
+            return DATA_TYPE_INTEGER
```

Comment:
non-zero sum could also fit in here since you may be summing something other than integers

Reply:
good call (made a small int since it's always 0 or 1). also changed average to a decimal which i think is more correct.